lib/JSON/Validator.pm

```diff
@@ -432,14 +446,12 @@ sub _resolve_ref {
     push @guard, $other->{'$ref'};
     confess "Seems like you have a circular reference: @guard" if @guard > RECURSION_LIMIT;
     last if !$ref or ref $ref;
-    $fqn = Mojo::URL->new($ref =~ m!^/! ? "#$ref" : $ref);
-    $fqn = $fqn->to_abs($url) unless $fqn->is_abs;
-    $url = $fqn;
-    $fqn = $fqn->to_string;
-    $fqn =~ s!(.)#$!$1!;
-    $fqn =~ s!#(.+)!{'#' . url_unescape $1}!e;
-
+    $fqn = $ref =~ m!^/! ? "#$ref" : $ref;
     ($base, $pointer) = split /#/, $fqn, 2;
+    $url = $base = _absolutise_location($base, $url);
```

Comment:
Not related to this PR, but I find my variables `$url` and `$base` confusing. You're free to rename them if you have a better suggestion.

Reply:
I've renamed `$base` to `$location` in line with your suggested rename of the absolute-ising function.